### PR TITLE
[W5.7][W16-3]Triston Pang Erh Syen

### PR DESCRIPTION
**Pull request by tristonpang**

### CODE REVIEW

**Comment by nelvintan** [comment]
### Testing
Some updates to the JUnit test or I/O test might help to ensure that your command is bug free.
 
### Other suggestions
Currently, you are sorting people based on the alphabetical order of their names. Instead you can try sorting them based on some other criteria. For example. the alphabetical order of their first names or tags if they exist.

**Comment by ngkangtze** [comment]
Hi Tristan,
Great code! Just a few comments: you should use the getter, toString(), to get the full name of the Person instead of calling the fullName attribute.
Also, there is no need to produce a duplicate of internalList for your sort function as each command generates a new UniquePersonList.
Cheers!

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,12 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Listing all persons alphabetically by name : `sort`
+
+Shows a list of all persons, along with their non-private details, in the address book alphabetically by name. +
+Format: `list`
+
+
```
Comment (joanneong) [inline docs/UserGuide.adoc:80]: Redundant blank line here :(

Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

Comment (joanneong) [reply docs/UserGuide.adoc:80]: Same for other areas in the code below!

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (joanneong) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Such import statements violate our coding standards...do you know why? How should this be instead? :P

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -81,6 +76,32 @@ public UniquePersonList(UniquePersonList source) {
...
+     * @param listToSort List of persons to be sorted
+     */
+    private void sortListOfPersons(List<Person> listToSort) {
+        listToSort.sort(new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return o1.getName().fullName.compareTo(o2.getName().fullName);
+            }
+        });
+    }
```
Comment (joanneong) [inline src/seedu/addressbook/data/person/UniquePersonList.java:103]: Good use of JavaDoc here :)

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 2 above)
Comment (joanneong) [reply src/seedu/addressbook/data/person/UniquePersonList.java:3]: Same for other areas in the code below!

**Review by joanneong — APPROVED** [review]
A decent attempt at the LO :)

However, it would be great if you can also write some tests for your enhancement (IO tests and JUnit tests), as @nelvintan mentioned as well.

Please close this PR after you read my (and their) comments!

**Pull request closed without merging**